docs/use_cases/embeddings_on_browser.md

```diff
@@ -0,0 +1,628 @@
+
+# Our step-by-step tutorial
+
+## Import modules
+
+1. **`import React, { FC, useState, useEffect, useCallback } from 'react';`**:
+    - This line imports necessary modules from the 'react' library.
+    - **`React`** is the core library for building user interfaces in React.
+    - **`FC`** is a TypeScript type definition, representing a functional component.
+    - **`useState`**, **`useEffect`**, and **`useCallback`** are React hooks that allow you to manage component state and side effects.
+2. **`import { Box, Grid, Typography, TextField, Paper, Button, CircularProgress } from "@mui/material";`**:
+    - This line imports various Material-UI components from the **`@mui/material`** library.
+    - Material-UI is a popular UI framework for React applications, providing pre-styled components for building user interfaces.
+3. **`import '@tensorflow/tfjs-backend-cpu';`** and **`import '@tensorflow/tfjs-backend-webgl';`**:
+    - These lines import the TensorFlow.js backend modules for CPU and WebGL.
+    - TensorFlow.js is a JavaScript library that allows machine learning and deep learning models to run in the browser.
+4. **`import * as use from '@tensorflow-models/universal-sentence-encoder';`**:
+    - This line imports the Universal Sentence Encoder model from TensorFlow.js.
+    - The Universal Sentence Encoder is a pre-trained model that encodes sentences into numerical vectors, allowing similarity comparisons between sentences.
+5. **`import * as tf from '@tensorflow/tfjs-core';`**:
+    - This line imports the core TensorFlow.js library.
+    - TensorFlow.js core provides fundamental functionalities for machine learning in JavaScript.
+6. **`import { interpolateGreens } from 'd3-scale-chromatic';`**:
+    - This line imports the **`interpolateGreens`** function from the 'd3-scale-chromatic' library.
+    - D3 is a data visualization library, and **`interpolateGreens`** is used to interpolate colors in the green color scale.
+
+```tsx
+import React, { FC, useState, useEffect, useCallback } from 'react';
+import {
+  Box,
+  Grid,
+  Typography,
+  TextField,
+  Paper,
+  Button,
+  CircularProgress,
+} from "@mui/material";
+
+import '@tensorflow/tfjs-backend-cpu';
+import '@tensorflow/tfjs-backend-webgl';
+
+import * as use from '@tensorflow-models/universal-sentence-encoder';
+import * as tf from '@tensorflow/tfjs-core';
+import { interpolateGreens } from 'd3-scale-chromatic';
+```
+
+## State variables to manage user input, loading state, and results
+
+These state variables are essential for managing user input, tracking loading states of machine learning models, and updating the user interface with the results, including the similarity matrix visualization.
+
+Here, we define a series of state variables using the **`useState`** hook in a React functional component. These state variables are used to manage various aspects of the component's behavior and user interface:
+
+1. **`sentences`** and **`setSentences`**:
```

Comment:
same here - just have commented code snippets and remove this separate commend sections